From c78bdf4f1e46d5b604063416a682470aa716e333 Mon Sep 17 00:00:00 2001
From: Sukrut Bellary <sbellary@baylibre.com>
Date: Thu, 22 Feb 2024 23:57:06 -0800
Subject: [PATCH 09/27] ARM: dts: DRA74x: Add the VIP3 node

The DRA74xx/AM57xx has three Video Input Ports (VIP) i.e., VIP1, VIP2,
and VIP3.
Add VIP3 port node which has 2 slices VIN5 and VIN6.
Forward port from tiL5.10 -> 6.1*.
This is a part of AM57 LTS migration activity.

Signed-off-by: Sukrut Bellary <sbellary@baylibre.com>
---
 arch/arm/boot/dts/dra74x.dtsi | 58 ++++++++++++++++++++++++++++++++++-
 1 file changed, 57 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/dra74x.dtsi b/arch/arm/boot/dts/dra74x.dtsi
index 03c2e2ecf610..34559be48713 100644
--- a/arch/arm/boot/dts/dra74x.dtsi
+++ b/arch/arm/boot/dts/dra74x.dtsi
@@ -49,7 +49,6 @@ dsp2_system: dsp_system@41500000 {
 			reg = <0x41500000 0x100>;
 		};
 
-
 		target-module@41501000 {
 			compatible = "ti,sysc-omap2", "ti,sysc";
 			reg = <0x41501000 0x4>,
@@ -293,4 +292,61 @@ vin4b: port@3 {
 			};
 		};
 	};
+
+	target-module@1b0000 {			/* 0x489b0000, ap 25 34.0 */
+		compatible = "ti,sysc-omap4", "ti,sysc";
+		reg = <0x1b0000 0x4>,
+		      <0x1b0010 0x4>;
+		reg-names = "rev", "sysc";
+		ti,sysc-midle = <SYSC_IDLE_FORCE>,
+				<SYSC_IDLE_NO>,
+				<SYSC_IDLE_SMART>;
+		ti,sysc-sidle = <SYSC_IDLE_FORCE>,
+				<SYSC_IDLE_NO>,
+				<SYSC_IDLE_SMART>;
+		clocks = <&cam_clkctrl DRA7_CAM_VIP3_CLKCTRL 0>;
+		clock-names = "fck";
+		#address-cells = <1>;
+		#size-cells = <1>;
+		ranges = <0x0 0x1b0000 0x10000>;
+
+		vip3: vip@0 {
+			compatible = "ti,dra7-vip3";
+			label = "vip3";
+			reg = <0x0000 0x114>,
+			      <0x5500 0xd8>,
+			      <0x5700 0x18>,
+			      <0x5800 0x80>,
+			      <0x5a00 0xd8>,
+			      <0x5c00 0x18>,
+			      <0x5d00 0x80>,
+			      <0xd000 0x400>;
+			reg-names = "vip",
+				    "parser0",
+				    "csc0",
+				    "sc0",
+				    "parser1",
+				    "csc1",
+				    "sc1",
+				    "vpdma";
+			interrupts = <GIC_SPI 353 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 394 IRQ_TYPE_LEVEL_HIGH>;
+			/* CTRL_CORE_SMA_SW_1 */
+			ti,vip-clk-polarity = <&scm_conf 0x534>;
+
+			ports {
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				vin5a: port@0 {
+					reg = <0>;
+					label = "vin5a";
+				};
+				vin6a: port@1 {
+					reg = <1>;
+					label = "vin6a";
+				};
+			};
+		};
+	};
 };
-- 
2.39.2

